### PR TITLE
Upgrade Android SDK to 33 Tiramisu

### PR DESCRIPTION
**Pull request by GaryQian**
Upgrades android SDK to 33 Tiramisu

Also adds a `list` subcommand to the sdk script for ease of use.

### CODE REVIEW

**Comment by GaryQian** [comment]
(artifacts are still uploading, tests will fail until uploading is complete in ~1-2hrs)

**Review by stuartmorgan — APPROVED** [review]
LGTM

File: tools/android_sdk/create_cipd_packages.sh [hunk 1]
```diff
@@ -69,10 +72,16 @@ while [ ! -f "$sdkmanager_path" ]; do
...
+# list available packages
+if [ $version_tag == "list" ]; then
+  $sdkmanager_path  --list --include_obsolete
+  exit 0
+fi
+
 # We create a new temporary SDK directory because the default working directory
 # tends to not update/re-download packages if they are being used. This guarantees
 # a clean install of Android SDK.
-temp_dir=`mktemp -d -t android_sdk`
```
Comment (GaryQian) [inline tools/android_sdk/create_cipd_packages.sh:75]: Linux requires XXXX characters to randomize

**Comment by chinmaygarde** [comment]
The test failures look related. Can we patch these and submit please?

**Comment by GaryQian** [comment]
The failures on TOT seem to be a flaky infra failure:
`fatal error in startup: could not invoke luciexe: launching luciexe: fork/exec /b/s/w/ir/kitchen-checkout/luciexe: no such file or directory`

I'm seeing it on this PR as well, but it goes away when re-run usually

**Pull request merged**